### DISCUSSION TITLE
Why is `recapture_schema` dangerous?

### DISCUSSION THREAD

**Discussion by nirdanon** [zendesk/maxwell#2045] [General]
Hi,

We've encountered a problem that maxwell tables list is missing some of the tables.
even after seeing the tables were created in maxwell log.

after restarting maxwell with the recapture_schema=true, the missing tables were added to maxwell tables list.

In the api docs it's mentioned as "dangerous"
https://maxwells-daemon.io/apidocs/com/zendesk/maxwell/MaxwellConfig.html#recaptureSchema

![image](https://user-images.githubusercontent.com/23293247/219422103-6bd029da-436a-489e-b665-4039a230a5ab.png)

can you explain why is it dangerous?
what is the risk of setting this?
in addition, why do we need to restart the process once this flag is set, in order to get the additional new tables?

thanks,
Nir

**Comment by osheroff** [comment]
the "current" schema may or may not match up to the schema represented by the binlog pointer; if your binlog pointer is behind in the replication stream you may end up with mis-matched schemas, or a situation where schema changes will play multiple times. 

Were you able to determine the cause of the missing tables?
